modules/Emails/include/ListView/ListViewHeader.js

```diff
@@ -37,18 +37,88 @@
  * display the words  "Powered by SugarCRM" and "Supercharged by SuiteCRM".
  */
 
+/**
+ * @todo take this function to a common email listview object
+ * @param {string} actionUrl
+ * @param {function} successCallback
+ * @param {function} errorCallback
+ * @param {string} loadingTitle
+ * @param {string} errorMessage
+ */
+var handleSelectedEmailsListViewItems = function(actionUrl, successCallback, errorCallback, loadingTitle, errorMessage) {
+
+  if(typeof loadingTitle === 'undefined') {
+    SUGAR.language.translate('Emails', 'LBL_LOADING');
+  }
+
+  if(typeof errorMessage === 'undefined') {
+    errorMessage = 'Error at selected emails handling.';
+  }
+
+  var mb = messageBox({backdrop: false});
+  mb.setTitle(loadingTitle);
+  mb.setBody('<div class="in-progress"><img src="themes/'+SUGAR.themes.theme_name+'/images/loading.gif"></div>');
+  mb.hideFooter();
+  mb.show();
+  mb.on('cancel', function() {
+    "use strict";
+    mb.remove();
+  });
+
+
+  var query = JSON.parse($('[name=MassUpdate] [name=current_query_by_page]').val());
+  var url = 'index.php?module=Emails&action=' + actionUrl;
+
+  var postOpts = {
+    "inbound_email_record": query.inbound_email_record,
+    "folders_id": query.folders_id,
+    "folder": query.folder,
+    "uid[]": []
+  };
+
+  if(document.MassUpdate.select_entire_list &&
+    document.MassUpdate.select_entire_list.value == 1) {
+    // Import all emails from mail box
+    postOpts.all = true;
+  } else {
+    postOpts.all = false;
+    // import only selected emails from inbox
+    $('.listview-checkbox').each(function(i,v) {
+      if($(v).is(':checked')) {
+        postOpts['uid[]'].push(query.email_uids[i]);
+      }
+    });
+  }
+
+  $.post( url, postOpts).done(function (data) {
+    var jsonData = JSON.parse(data);
+    mb.hide();
+    if(jsonData.response) {
+      successCallback(jsonData);
+    } else {
+      if(errorCallback) {
+        errorCallback(jsonData);
+      } else {
+        console.error(errorMessage + ' Please check the logs for details.');
+      }
+    }
+  }).error(function (data) {
+    mb.hide();
+    console.error(data);
+  });
+};
+
 $(document).ready(function () {
 
   var query = JSON.parse($('[name=current_query_by_page]').val());
+  var $btnEmailsCurrentFolder = $('.btn-emails-current-folder');
+
   if(typeof query.folder === 'undefined' ||  query.folder === '') {
-    $('.btn-emails-current-folder').remove();
+    $btnEmailsCurrentFolder.remove();
```

Comment:
Missing () perhaps. Please do not use $variable for varable names. The $ sign signifies dependency injection in MVVM frameworks like angular. Plus it looks like broken jQuery.